src/App.js

```diff
@@ -1,26 +1,77 @@
 
-import React, { useEffect, useReducer } from 'react';
-import TodoList from './TodoList'
-import AddTodoForm from './AddTodoForm'
-import {
-  BrowserRouter, Routes, Route
-} from "react-router-dom";
+import { useEffect, useReducer, useState } from 'react';
+import {BrowserRouter, Routes, Route} from "react-router-dom";
+import TodoList from './components/TodoList'
+import AddTodoForm from './components/AddTodoForm'
+import SpeechText from './components/SpeechText'
+import Search from './components/Search'
+import Pagination from './components/Pagination'
+import ThemeToggle   from './ThemeToggle'
+import ThemeContext from './ThemeContext'
+import styles from './static/App.module.css'
+import "./index.css";
+import paths from './paths'
+
 
 function App() {
-  const defaultHeaders = {
-    Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
-  }
+  const [isDarkMode, setIsDarkMode] = useState(true)
+  const [sortTitle, setSortTitle] = useState(true)
+  const [sortTime, setSortTime] = useState(true)
+  const [currentPage, setCurrentPage] = useState(1)
+  const [todoPerPage] = useState(10)
 
+  function toggleTheme () {
+    setIsDarkMode((prevIsDarkMode) => !prevIsDarkMode)
+  }
+  
 
 
+  const defaultHeaders = {
+    Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+  }
   const stateManagementFunction = (previousState, action) => {
     switch (action.type) {
+      case 'START_LOADING_TITLES':
+        return {
+          ...previousState,
+          isLoading: true,
+        }
       case 'FINISH_LOADING_TITLES':
         return {
           ...previousState,
           isLoading: false,
           todoList: action.payload.todoList
         }
+      case 'SORT_TITLES_AZ':
+        return {
+            ...previousState,
+            isLoading: false,
+            todoList: previousState.todoList.sort((a,b)=> a.title.localeCompare(b.title))
+          }
+
+      case 'SORT_TITLES_ZA':
```

Comment:
this whole statemangement function can be in another file/module that just exports the function.